axelrod/strategies/titfortat.py

```diff
@@ -273,3 +273,59 @@ def reset(self):
         Player.reset(self)
         self.randomness_counter = 0
         self.deadlock_counter = 0
+
+
+class Gradual(Player):
+    """
+    A player that punishes defections with a growing number of defections
+    but after punishing enters a calming state and cooperates no matter what
+    the opponent does for two rounds.
+
+    http://perso.uclouvain.be/vincent.blondel/workshops/2003/beaufils.pdf """
+
+    name = "Gradual"
+    classifier = {
+        'memory_depth': float('inf'),
+        'stochastic': False,
+        'makes_use_of': set(),
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+
+    calming = False
```

Comment:
Also, these four initialization lines should be in `def __init__(self)` since they can differ for multiple instances of the class. When variables are defined at the class level they are shared by all instances of the class.

In other words, we need an `__init__` function that does just what you've done in `reset` already:

```
def __init__(self):
    Player.__init__(self)
    self.calming = False
    self.punishing = False
    self.punishment_count = 0
    self.punishment_limit = 0
```